src/screens/WelcomeScreen.js

```diff
@@ -52,27 +130,128 @@ class WelcomeScreen extends Component {
-          <Image
-            style={{resizeMode: 'cover', height: 207, width: 189}}
-            source={Logo}></Image>
-          <View style={{marginTop: 28}}>
+        <Modal
+          animationType="slide"
+          transparent={true}
+          visible={this.state.isVisible}
+          onRequestClose={() => {
+            Alert.alert('Modal has been closed.');
+          }}>
+          <View
+            style={{
+              flex: 1,
+              justifyContent: 'center',
+              alignItems: 'center',
+            }}>
+            <View
+              style={{
+                borderRadius: 15,
+                backgroundColor: '#6BB1F1',
+                height: 346,
+                width: 370,
+              }}>
+              <View style={{flex: 0.15, alignItems: 'flex-end'}}>
+                <View style={{marginRight: 15, marginTop: 15}}>
+                  <TouchableOpacity onPress={() => this.displayModal(false)}>
+                    <AntDesign name="close" size={30}></AntDesign>
+                  </TouchableOpacity>
+                </View>
+              </View>
+              <View
+                style={{
+                  flex: 0.75,
+                  alignItems: 'center',
+                }}>
+                <View style={{flex: 0.1}}>
+                  <Text style={{fontWeight: 'bold', fontSize: 24}}>
+                    Settings
+                  </Text>
+                </View>
+                <View
+                  style={{
+                    padding: 16,
+                    flex: 0.9,
+                    flexDirection: 'row',
+                  }}>
+                  <View style={{flex: 1}}>
+                    <Text style={{fontSize: 24, fontWeight: 'bold'}}>
+                      Category:{' '}
+                      <RNPickerSelect
+                        disabled={this.state.settingsDisabled}
+                        style={{fontSize: 24, fontWeight: 'bold'}}
+                        onValueChange={(value) =>
+                          this.setState({category: value})
+                        }
+                        items={modalCategories}
+                      />
+                    </Text>
+                    <Text
+                      style={{marginTop: 10, fontSize: 24, fontWeight: 'bold'}}>
+                      Difficulty:{' '}
+                      <RNPickerSelect
+                        disabled={this.state.settingsDisabled}
+                        style={{fontSize: 24, fontWeight: 'bold'}}
+                        onValueChange={(value) =>
+                          this.setState({difficulty: value})
+                        }
+                        items={modalDifficulty}
+                      />
+                    </Text>
+                  </View>
+                </View>
+              </View>
+              <View style={{flex: 0.1}}></View>
+            </View>
+          </View>
+        </Modal>
+        <View style={({flex: 0.9}, mainContainer)}>
```

Comment:
` <View style={...{flex: 0.9}, ...styles.mainContainer}>`
 boyle olmasi gerekiyor, yaptigin sekilde olunca calismamasi gerekiyor bence. 

https://laska.io/blog/combining-multiple-styles-in-react-native/
bunu incelersin